rich_clipboard_macos/macos/Classes/RichClipboardPlugin.swift

```diff
@@ -36,26 +36,23 @@ public class RichClipboardPlugin: NSObject, FlutterPlugin {
         }
         return result
     }
-
+    
     func setData(_ arguments: Any?) {
-        guard let data = arguments as? [String: String] else {
-            return
-        }
-
-        guard let text = data[mimeTextPlain] else {
+        guard let data = arguments as? [String: String?] else {
```

Comment:
This comment was supposed to be in my previous review, but Github ate it or something.

Rather than account for `nil` everywhere this dictionary is accessed, we can just filter the `nil` values out here. This is already implemented in the iOS plugin:

```suggestion
        guard let data = (arguments as? [String: String?])?.filter({ $0.value != nil }) as? [String: String] else {
```

Alternatively, there's a way to do the same thing with less code and one less typecast using [`compactMapValues`][2]:

```suggestion
        guard let data = (arguments as? [String: String?])?.compactMapValues({ $0 }) else {
```

I'm good with either approach. If you go with the second option please also update [the matching line in the iOS plugin][2] so we're consistent across the Swift portions of the codebase.

[1]: https://github.com/BringingFire/rich_clipboard/blob/3ad6510f7bce215ea0f846450b5b35e1b1772678/rich_clipboard_ios/ios/Classes/RichClipboardPlugin.swift#L56
[2]: https://developer.apple.com/documentation/swift/dictionary/compactmapvalues(_:)

Reply:
@jmatth Thanks for the review. These are really good points. I'll update the code.